### PR TITLE
Quick hull

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
# PR Description

This Pull Request provides an implementation of the Quick Hull convex hull algorithm. It works in arbitrary dimension. It provides several kernels to deal with lattice or rational points, and also to compute the Delaunay cell complex.

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+To compute the convex hull of lattice points, you need to include `QuickHull.h`.
+
+\snippet geometry/tools/exampleLatticeBallQuickHull3D.cpp QuickHull3D-Includes
+
+You should then define some typedefs:
+
+\snippet geometry/tools/exampleLatticeBallQuickHull3D.cpp QuickHull3D-Typedefs
+
+Then we assume that \a V contains a range of lattice 3D points (with
+either 32 bits or 64 bits precision). You may compute their convex
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:120]: can the Integer type be "BigInteger"?

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+   - reassign "above" points of \a f to these new facets and put them into the queue
+
+3. The algorithm stops when all current facets have no "above" points.
+
+4. Optionnaly extract vertices of the convex hull polytopes by
+determining the "on" points that belong to at least \a d facets.
+
+@note In opposition with the usual `qhull` implementation, this
+class uses a kernel that can be chosen in order to provide exact
+computations. This is the case for lattice points, but also rational
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:66]: for fixed precision integers, (cf comment below on BigInteger)?

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+coordinates. Real points with `double` precision are rounded to these
+nearest rational coordinates. 
+
+@note Different real points may be rounded to the same rational point.
+
+You may check computation times with (QuickHull::timings)
+
+\snippet geometry/tools/exampleRationalBallQuickHull3D.cpp QuickHull3D-Timings
+
+You can easily retrieves the rational vertex positions with
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:200]:
```suggestion
You can easily retrieve the rational vertex positions with
```

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+total time      = 7882 ms.
+\endverbatim
+ 
+@note Half of the time is spent on removing duplicated input
+points. If you know that your range of points does not contain
+duplicates, you can specify it to QuickHull::setInput.
+
+@note If you choose a too big precision, integer computations required
+by convex hull computations might overflow. With this kernel, the
+internal type for integers is 64 bits precision. Hence the higher the
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:251]: ok this could be clarified earlier I think

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+hull (facets and vertices) with (QuickHull::setInput and QuickHull::computeConvexHull):
+
+\snippet geometry/tools/exampleLatticeBallDelaunay2D.cpp Delaunay2D-Computation
+
+You can easily retrieves the vertex positions with
+QuickHull::getVertexPositions (you can put them in a vector of real or
+lattice points), and retrieves the consistently oriented vertices for
+each \b cell with QuickHull::getFacetVertices.
+
+In opposition with convex hull computation, facets here designate the
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:272]:
```suggestion
In opposition to convex hull computation, facets here designate the
```
?

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+
+Building a lattice polytope in nD just requires a call to
+ConvexityHelper::computeLatticePolytope.
+
+\code
+std::vector<Z2i::Point> V = { Point(0,0), Point(-4,-1), Point(-3,5), Point(7,3), Point(5, -2) };
+const auto P = ConvexityHelper< 2 >::computeLatticePolytope( V );
+\endcode
+
+@note For dimension up to 3D, you may precise if you wish your lattice
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:362]:
```suggestion
@note For dimension up to 3D, you may indicate if you wish your lattice
```

File: src/DGtal/geometry/doc/moduleQuickHull.dox [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+You may simply use one of the two static methods
+ConvexityHelper::computeConvexHullBoundary to build either a
+SurfaceMesh or a PolygonalSurface that represents the boundary convex
+hull of a set of 3D real points. The precision is given as a scale
+factor that transforms real coordinates before rounding them at the
+nearest integer.
+
+@see exampleQuickHull3D.cpp
+
+@note In opposition with ConvexityHelper::computeRationalPolytope the
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleQuickHull.dox:441]:
```suggestion
@note In opposition to ConvexityHelper::computeRationalPolytope the
```
?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 8]
```diff
@@ -0,0 +1,1599 @@
...
+#include "DGtal/geometry/tools/QuickHullKernels.h"
+
+namespace DGtal
+{
+  /////////////////////////////////////////////////////////////////////////////
+  // template class QuickHull
+  /**
+     Description of template class 'QuickHull' <p> \brief Aim:
+     Implements the quickhull algorithm, a famous arbitrary
+     dimensional convex hull computation algorithm. It relies on
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:59]: add cite

File: src/DGtal/geometry/tools/QuickHull.h [hunk 9]
```diff
@@ -0,0 +1,1599 @@
...
+#include <set>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/Clock.h"
+#include "DGtal/geometry/tools/QuickHullKernels.h"
+
+namespace DGtal
+{
+  /////////////////////////////////////////////////////////////////////////////
+  // template class QuickHull
+  /**
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:56]: doxygen generates "comment" blocks for this doc (because of the four " "before each line maybe). 
remove the spaces?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 10]
```diff
@@ -0,0 +1,1599 @@
...
+  // (4) output result as OBJ file
+  std::ofstream out( "qhull.obj" );
+  DGtal::SurfaceMeshWriter< Space::RealPoint, Space::RealVector >
+    ::writeOBJ( out, mesh );
+  out.close();
+  return 0;
+} 
+@endcode
+
+     @note In opposition with the usual QuickHull implementation, this
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:116]: same here.. doxygen interprets the spaces as "comment block"

File: src/DGtal/geometry/tools/QuickHull.h [hunk 11]
```diff
@@ -0,0 +1,1599 @@
...
+    typedef std::size_t                Index;
+    typedef std::size_t                Size;
+    BOOST_STATIC_ASSERT(( Point::dimension == Vector::dimension ));
+    typedef std::vector< Index >       IndexRange;
+    typedef typename Kernel::HalfSpace HalfSpace;
+    typedef typename Kernel::CombinatorialPlaneSimplex CombinatorialPlaneSimplex;
+    static const Size  dimension  = Point::dimension;
+
+    // BOOST_STATIC_CONSTANT( Index, UNASSIGNED = (Index) -1 );
+    // static constexpr Index UNASSIGNED = (Index) -1;
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:150]: clean?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 12]
```diff
@@ -0,0 +1,1599 @@
...
+      void display( std::ostream& out ) const
+      {
+        const Vector N = H.internalNormal(); 
+        out << "[Facet iN=(" << N[0];
+        for ( Dimension i = 1; i < N.dimension; i++ ) out << "," << N[ i ];
+        out << ") c=" << H.internalIntercept() << " b=" << below << " n={";
+        for ( auto&& n : neighbors ) out << " " << n;
+        out << " } #out=" << outside_set.size();
+        // out << " } out={";
+        // for ( auto&& n : outside_set ) out << " " << n;
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:194]: clean?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 13]
```diff
@@ -0,0 +1,1599 @@
...
+    /// otherwise.
+    bool getFacetHalfSpaces( std::vector< HalfSpace >& facet_halfspaces )
+    {
+      facet_halfspaces.clear();
+      if ( ! ( status() >= Status::FacetsCompleted
+               && status() <= Status::AllCompleted ) ) return false;
+      facet_halfspaces.reserve( nbFacets() );
+      for ( Index f = 0; f < nbFacets(); ++f )  {
+        facet_halfspaces.push_back( facets[ f ].H );
+          // kernel.normal(  ),
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:697]: clean?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 14]
```diff
@@ -0,0 +1,1599 @@
...
+    /// point index -> vertex index (or UNASSIGNED)
+    IndexRange p2v;
+    /// vertex index -> point index
+    IndexRange v2p;
+    /// Number of finite facets
+    Size nb_finite_facets;
+    /// Number of infinite facets (!= 0 only for specific kernels)
+    Size nb_infinite_facets;
+    /// Timings of the different phases: 0: init, 1: facets, 2: vertices.
+    std::vector< double > timings;
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:827]: std::array<double, 3> ?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 15]
```diff
@@ -0,0 +1,1599 @@
...
+    }
+  
+    /// @return an unused facet index
+    Index newFacet()
+    {
+      // SLightly faster to postpone deletion of intermediate facets.
+      const Index f = facets.size();
+      facets.push_back( Facet() );
+      return f;
+      // if ( ! deleted_facets.empty() ) {
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:1215]: deadcode

File: src/DGtal/geometry/tools/QuickHull.h [hunk 16]
```diff
@@ -0,0 +1,1599 @@
...
+    {
+      ASSERT( if1 != if2 );
+      const Facet& f1 = facets[ if1 ];
+      const Facet& f2 = facets[ if2 ];
+      if ( kernel.equal( f1.H, f2.H ) ) return true;
+      // Need to check if one N is a multiple of the other.
+      for ( auto&& v : f1.on_set )
+        if ( ! on( f2, points[ v ] ) ) return false;
+      // No need to double check.
+      // for ( auto&& v : f2.on_set )
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:1294]: keep that one for doc ?

File: src/DGtal/geometry/tools/QuickHull.h [hunk 17]
```diff
@@ -0,0 +1,1599 @@
...
+    IndexRange pointsOnRidge( const Ridge& R ) const
+    {
+      // Slightly faster to look at points instead at true geometry.
+      IndexRange result;
+      const Facet& f1 = facets[ R.first ];
+      const Facet& f2 = facets[ R.second ];
+      std::set_intersection( f1.on_set.cbegin(), f1.on_set.cend(),
+                             f2.on_set.cbegin(), f2.on_set.cend(),
+                             std::back_inserter( result ) );
+      // if ( f1.on_set.size() <= f2.on_set.size() ) {
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHull.h:1350]: deadcode

File: src/DGtal/geometry/tools/QuickHullKernels.h [hunk 18]
```diff
@@ -0,0 +1,805 @@
...
+    void makeInput( std::vector< Point >& processed_points,
+                    IndexRange& input2comp, IndexRange& comp2input,
+                    const std::vector< InputPoint >& input_points,
+                    bool remove_duplicates )
+    {
+      const auto F = [&] ( InputPoint input ) -> Point
+        {
+          Point p;
+          for ( Dimension i = 0; i < dimension; i++ )
+            p[ i ] = static_cast< Scalar >( round( input[ i ] * precision ) );
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHullKernels.h:649]: This conversion formula should  probably be more explicit in the class/function description (relying on rounding on doubles)

File: src/DGtal/geometry/tools/QuickHullKernels.h [hunk 19]
```diff
@@ -0,0 +1,805 @@
...
+                    IndexRange& input2comp, IndexRange& comp2input,
+                    const std::vector< InputPoint >& input_points,
+                    bool remove_duplicates )
+    {
+      const auto F = [&] ( InputPoint input ) -> Point
+        {
+          Point p;
+          Scalar z = 0;
+          for ( Dimension i = 0; i < dimension - 1; i++ ) {
+            const Scalar x = static_cast< Scalar >( round( input[ i ] * precision ) );
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHullKernels.h:775]: cf earlier comment (and I will skip the next ones)

File: src/DGtal/geometry/volumes/ConvexityHelper.h [hunk 20]
```diff
@@ -0,0 +1,424 @@
...
+  /////////////////////////////////////////////////////////////////////////////
+  // template class ConvexityHelper
+  /**
+     Description of template class 'ConvexityHelper' <p> \brief Aim:
+     Provides a set of functions to facilitate the computation of
+     convex hulls and polytopes, as well as shortcuts to build cell
+     complex representing a Delaunay complex.
+
+     @tparam dim the dimension of the space where points and further objects live.
+  */
```
Comment (dcoeurjo) [inline src/DGtal/geometry/volumes/ConvexityHelper.h:65]: missing doc for TInteger type (and this type is not "concept checked")

**Review by dcoeurjo — COMMENTED** [review]
Great PR (and very nice illustrations).
I do have few suggestions.

**Comment by dcoeurjo** [comment]
`testDigitalConvexity` is a bit long (26sec)

File: examples/geometry/tools/exampleLatticeBallDelaunay2D.cpp [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+  Delaunay2D hull;
+  hull.setInput( V );
+  hull.computeConvexHull();
+  std::cout << "#points="    << hull.nbPoints()
+            << " #vertices=" << hull.nbVertices()
+            << " #facets="   << hull.nbFacets() << std::endl;
+  //! [Delaunay2D-Computation]
+  //! [Delaunay2D-Timings]
+  double total_time = 0;
+  std::for_each( hull.timings.cbegin(), hull.timings.cend(),
```
[the remark this answers is not in the document]
Comment (rolanddenis) [reply examples/geometry/tools/exampleLatticeBallDelaunay2D.cpp:89]: If I'm not wrong, it exists in previous language versions also, but it has been updated in C++17 (execution policy) and C++20 (`constexpr`).

[the remark this answers is not in the document]
Comment (dcoeurjo) [reply examples/geometry/tools/exampleLatticeBallDelaunay2D.cpp:89]: Correct.. i read too quickly the "since 17" in https://en.cppreference.com/w/cpp/algorithm/for_each

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 1 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:120]: not in this version, I am afraid

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 3 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:200]: done

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 5 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:272]: done

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 6 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:362]: done

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 7 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:441]: done

File: src/DGtal/geometry/tools/QuickHull.h (hunk 8 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:59]: done

File: src/DGtal/geometry/tools/QuickHull.h (hunk 9 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:56]: done

File: src/DGtal/geometry/tools/QuickHull.h (hunk 10 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:116]: done

File: src/DGtal/geometry/tools/QuickHull.h (hunk 11 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:150]: ok

File: src/DGtal/geometry/tools/QuickHull.h (hunk 12 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:194]: done

File: src/DGtal/geometry/tools/QuickHull.h (hunk 13 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:697]: ok

File: src/DGtal/geometry/tools/QuickHull.h (hunk 14 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:827]: Sometimes you do not process till the vertices, but stop at facets, so only two timings are computed.

File: src/DGtal/geometry/tools/QuickHull.h (hunk 15 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:1215]: ok

File: src/DGtal/geometry/tools/QuickHull.h (hunk 16 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:1294]: cleaning

File: src/DGtal/geometry/tools/QuickHull.h (hunk 17 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHull.h:1350]: cleaning

File: src/DGtal/geometry/tools/QuickHullKernels.h (hunk 18 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHullKernels.h:649]: Doc added here and in the kernel class to make more explicit the conversions.

File: src/DGtal/geometry/tools/QuickHullKernels.h (hunk 19 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHullKernels.h:775]: Doc added here and in the kernel class to make more explicit the conversions.

File: src/DGtal/geometry/volumes/ConvexityHelper.h (hunk 20 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/volumes/ConvexityHelper.h:65]: done

**Comment by JacquesOlivierLachaud** [comment]
Fixed codacy issues

**Comment by JacquesOlivierLachaud** [comment]
And tried to speed-up testDigitalConvexity

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 1 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:151]: Ok I have updated the four kernels. Now it handle BigInteger for internal computations and also for lattice points.

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 2 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox:66]: Now you have the choice of integer representation

File: src/DGtal/geometry/doc/moduleQuickHull.dox (hunk 4 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleQuickHull.dox]: This is now explained at the beginning of section 2

**Comment by JacquesOlivierLachaud** [comment]
All done !

File: src/DGtal/geometry/tools/QuickHullKernels.h [hunk 22]
```diff
@@ -0,0 +1,1315 @@
...
+    //       /// @return the same point
+    //       static
+    //       PointVector< dim, DGtal::int64_t >
+    //       cast( PointVector< dim, DGtal::int64_t > p )
+    //       {
+    //         return p;
+    //       }
+      
+    // #ifdef WITH_BIGINTEGER
+    //       /// @param i any integer
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHullKernels.h:281]: why this code is commented? not working?

File: src/DGtal/kernel/IntegerConverter.h [hunk 23]
```diff
@@ -0,0 +1,395 @@
...
+    cast( PointVector< dim, Integer > p )
+    {
+      return p;
+    }
+  };
+
+  /// Allows seamless conversion of integral types and lattice
+  /// points, while checking for errors when going from a more
+  /// precise to a less precise type.
+  ///
```
Comment (dcoeurjo) [inline src/DGtal/kernel/IntegerConverter.h:143]: For all this, why not relying on NumericalTraits?

Comment (dcoeurjo) [reply src/DGtal/kernel/IntegerConverter.h:143]: Maybe this is introducing duplicate codes/features

**Review by dcoeurjo — COMMENTED** [review]
Hard to follow, github has lot the "edits after the last review"

**Comment by dcoeurjo** [comment]
I'd need to clone locally. Is the new version working on big integers without floating point computations?

File: src/DGtal/kernel/IntegerConverter.h (hunk 23 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/kernel/IntegerConverter.h:143]: Because they are not sufficient for this purpose.
In fact gmp cannot initialize a big int from an int64 variable (but can from a const expression).
Conversely, gmp can do some conversion with long (which may or may not be 64 bits), but not on long long.
Finally, I wish to print warnings when bad conversions occur.

**Comment by JacquesOlivierLachaud** [comment]
The new version normally works with int32, int64, BigInteger (coordinate or internal integers).
In checkLatticeBallQuickHull, I tested from 2D to 6D, pairs int64/int64, int64/BigInteger, BigInteger/BigInteger

File: src/DGtal/kernel/IntegerConverter.h (hunk 23 above)
Comment (dcoeurjo) [reply src/DGtal/kernel/IntegerConverter.h:143]: should we remove the NumberTraits converter and use this one?

Comment (JacquesOlivierLachaud) [reply src/DGtal/kernel/IntegerConverter.h:143]: Perhaps we have to make more checks before doing that. Anyway the numbertraits converter just converts to int64. I was afraid that it was incorrect for BigInteger but it was working on my mac (where long is also 64 bits). So perhaps it is no harm keeping it. Otherwise, a cleaner IntegerConverter would have a "status", instead of displaying warnings. We can perhaps make an issue and I do a small PR afterwards to do that. What do you think ? Among other limitations, I only treat the signed case, and I do not care about small integers.

Comment (dcoeurjo) [reply src/DGtal/kernel/IntegerConverter.h:143]: I don't know.. I am afraid that having converters at several location could be a problem (like the dozen of functors/classes to threshold an image). For me, NumberTraits is a good place to contain such helpers/converters. Let me check again the converters you have here...

File: src/DGtal/geometry/tools/QuickHullKernels.h [hunk 24]
```diff
@@ -0,0 +1,1315 @@
...
+#include "DGtal/kernel/NumberTraits.h"
+#include "DGtal/kernel/PointVector.h"
+#include "DGtal/kernel/IntegerConverter.h"
+#include "DGtal/math/linalg/SimpleMatrix.h"
+
+namespace DGtal
+{
+  namespace detail {
+
+    // #ifdef WITH_BIGINTEGER
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/QuickHullKernels.h:58]: Here

Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/tools/QuickHullKernels.h]: You are right. Done.

**Comment by JacquesOlivierLachaud** [comment]
Should be ok now ;-)

**Comment by dcoeurjo** [comment]
Thx!

**Review by dcoeurjo — APPROVED** [review]

**Comment by dcoeurjo** [comment]
🎉🎉🎉

**Comment by dcoeurjo** [comment]
I think we are good for the release (plenty of new stuff + python)

**Comment by JacquesOlivierLachaud** [comment]
Yes. DGtal is very alive !
